packages/fuels-contract/src/script.rs

```diff
@@ -405,6 +408,89 @@ impl Script {
     }
 }
 
+/// Based on the returned Contract's output_params and the receipts returned from a call,
+/// decode the values and return them.
+pub fn get_decoded_output(
+    output_param: ParamType,
+    receipts: &mut Vec<Receipt>,
+) -> Result<Token, Error> {
+    // Multiple returns are handled as one `Tuple` (which has its own `ParamType`)
+    let (encoded_value, index) = match output_param.get_return_location() {
+        ReturnLocation::ReturnData => {
+            match receipts.iter().find(|&receipt| {
+                matches!(receipt,
+                    Receipt::ReturnData { id: _, data, .. } if !data.is_empty())
+            }) {
+                Some(r) => {
+                    let index = receipts.iter().position(|elt| elt == r).unwrap();
+                    (r.data().unwrap().to_vec(), Some(index))
+                }
+                None => (vec![], None),
+            }
+        }
+        ReturnLocation::Return => {
+            match receipts
+                .iter()
+                .find(|&receipt| matches!(receipt, Receipt::Return { .. }))
+            {
+                Some(r) => {
+                    let index = receipts.iter().position(|elt| elt == r).unwrap();
+                    (r.val().unwrap().to_be_bytes().to_vec(), Some(index))
+                }
+                None => (vec![], None),
+            }
+        }
+    };
+    if let Some(i) = index {
+        receipts.remove(i);
+    }
+    let decoded_value = ABIDecoder::decode_single(&output_param, &encoded_value)?;
+    Ok(decoded_value)
+}
+
+/// Run the script binary located at `binary_filepath` and return its resulting receipts,
+/// without having to setup a node or contract bindings.
+pub async fn run_script_binary(
```

Comment:
Since there is no wallet involved, the tx is transmitted without any inputs to cover the fee. This will work as long as the tx parameters are set to default (gas cost = 0). However, if the user supplies their own parameters, this may fail .

Reply:
Oh yeah I just realized it's super ill-designed… I think the refactor fixes this but not even sure lol. I'm going to test this.

Reply:
Hmm... then I believe this should be supported in _this_ PR, not in a subsequent one; No proper scenario will use parameters like gas cost set to zero.

Reply:
Yes I'm going to refactor and test this inside this PR